designs/ui-module/README.md

```diff
@@ -0,0 +1,261 @@
+
+3. Dashboard layout editor  
+Corresponds to the UI Module layout.  
+The size of UI module is fixed and cannot be changed.  
+Layout editor screen example when UI_SUBGROUP is included:  
+![layout-editor](./layout-editor.png)
+
+4. UI module layout editor  
+Layout editor to edit the layout of UI module.  
+The layout of UI module can be edited by pressing the [Edit Subgroup Layout] button from the edit screen of UI_SUBGROUP_T template node.  
+The size of UI module can be edited within the width and height of UI_SUBGROUP_T template node.  
+The width on the editor is fixed. Also, editing the layout is limited to the height specified by the size.  
+The edited result is reflected in the array value of "widgetOrder" of UI_SUBGROUP_T node.  
+SUBGROUP layout editor screen:  
+![subgroup-tmpl-layout](./subgroup-tmpl-layout.png)  
+*It is possible to support the layout editing function using the tree UI in dashboard layout sidebar, but here we will consider support separately.
+
+5. Dashboard layout sidebar  
+In dashboard layout sidebar, UI_SUBGROUP_I instance can be placed under the group.  
+Widget contained in UI_SUBGROUP_I instance is not displayed, only UI_SUBGROUP_I instance is displayed.  
+Pressing the [edit] button opens the edit screen for UI_SUBGROUP_I instance.  
+Layout Tabs & Lynks editor:  
+![layout-tab-link](./layout-tab-link.png)   
+*UI_SUBGROUP_I instance settings can be edited with the [edit] button on this screen.  
+To edit the layout of UI module(Template), open the UI_SUBGROUP_T template edit screen from the instance screen and press the [Edit Subgroup Layout] button to open the layout screen.
+
+### UI module function operation procedure
+
+The operation procedure when the user arranges the UI module on the dashboard is shown below.  
+
+1. Create SUBFLOW template  
+The user creates SUBFLOW template and places widget.
+
+2. Edit UI_SUBGROUP_T template  
+In step 1, Node-RED dashboard automatically creates UI_SUBGROUP_T template.  
+The user can edit the layout of UI module by opening the edit screen of UI_SUBGROUP_T template node from the node "info" tab.  
+
+3. Create SUBFLOW instance  
+Place SUBFLOW instance on tab or on SUBFLOW template.
+
+4. Edit UI_SUBGROUP_I instance  
+In step 3, Node-RED dashboard automatically generates UI_SUBGROUP_I instance.  
+The user opens the edit screen of UI_SUBGROUP_I instance node from the node "info" tab, etc., and specifies UI_GROUP of arbitrary Node-RED dashboard.  
+However, if you place it on SUBFLOW template, you do not need to specify UI_GROUP.
+
+5. Layout of dashboard  
+The user edits the layout of dashboard that contains the UI module.  
+UI module size is fixed.  
+
+6. SUBFLOW property support(optional)  
+If UI_GROUP is defined as the environment variable for the property definition of SUBFLOW, UI_GROUP of Node-RED dashboard can be specified in step 3.  
+In this case, the procedure for setting UI_GROUP in step 4 can be omitted. (See below)
+
```

Comment:
Can you explain why this is needed? Why do we need two different ways for the UI_GROUP to be set? Is there a particular scenario it addresses?

Reply:
Thank you for checking.
By using _UI_GROUP, I would like to be able to use the UI Module with the same operation as a widget.
For UI Module users, it's easy to understand because it can be treated like a widget.
![image](https://user-images.githubusercontent.com/49051279/103064009-3ba6f180-45f6-11eb-99e2-2c5be404eb6f.png)
However, the creator of the UI Module (subflow) must add the _UI_GROUP property to the subflow.
The subflow also needs the ability to set the Config node in the property.
Therefore, it is proposed as an optional function.